drivers/sensor/bh1749/bh1749.c

```diff
@@ -219,8 +217,8 @@ static void bh1749_async_init(struct k_work *work)
 			data->ready = true;
 			LOG_INF("BH1749 initialized");
 		} else {
-			k_delayed_work_submit(&bh1749_init_work,
-					      K_MSEC(async_init_delay[data->async_init_step]));
+			k_work_schedule(&bh1749_init_work,
```

Comment:
Use of schedule here seems OK, but I'll note that Zephyr is unprepared to correctly handle asynchronous device initialization.  The driver will successfully "initialize" if the submit of this work item succeeds; if the initialization fails up on line 211 it seems that all subsequent uses of the device will return `-EBUSY`, with no indication that the device initialization will never complete.

Reply:
@anangl anything to do here?

Reply:
No, at least not in this PR.

Reply:
Then we need to merge this one :)